formbuilder/components/builder/FieldListDropdown.js

```diff
@@ -5,34 +5,43 @@ import {Dropdown, MenuItem}  from "react-bootstrap"
 
 import config from "../../config";
 
-export default class AddFieldDropdown extends Component {
+export default class FieldListDropdown extends Component {
   constructor(props) {
     super(props);
-    this.state = {"fieldList": config.fieldList};
+    this.state = {
+      fieldList: config.fieldList
+    };
   }
 
-  handleAddField(fieldIndex, event) {
+  handleFieldListAction(fieldIndex, event) {
     const fieldList = this.state.fieldList;
+
     fieldIndex = parseInt(fieldIndex, 10);
 
     if (typeof fieldList[fieldIndex] != "undefined") {
-      this.props.addField(fieldList[fieldIndex]);
+      const field = fieldList[fieldIndex];
+
+      if (this.props.currentFieldName !== false) {
```

Comment:
It's not clear to me here what it means when `currentfieldName` is set to `false`. We probably should use an intermediary variable here to make it clearer.